sktime/datatypes/_convert.py

```diff
@@ -115,6 +132,16 @@ def convert(obj, from_type: str, to_type: str, as_scitype: str = None, store=Non
         as_scitype = mtype_to_scitype(to_type)
     elif not isinstance(as_scitype, str):
         raise TypeError("as_scitype must be str or None")
+    if store is not None and not isinstance(store, dict):
+        raise TypeError("store must be a dict or None")
+    if store_behaviour is not None and not isinstance(store_behaviour, str):
+        raise TypeError("store_behaviour must be a str or None")
+    elif store_behaviour not in [None, "reset", "freeze", "update"]:
```

Comment:
why is this `elif`? Can this not just be `if`?

Reply:
I guess it could be, does it make a difference?
The first two lines could even be removed.
Although the current way is perhaps more formulaic: first check type, then continuing from that check values, so perhaps easier to read.

Reply:
I simplified the input check, have a look